File: Normal/2230/main.py
from collections import deque


class MinStack:
    stack: deque[int] 
    minStack: deque[int] 

    def __init__(self):
        self.stack = deque([])
        self.minStack = deque([])


    def push(self, x: int) -> None:
        if len(self.stack) == 0:
            self.minStack.append(x)
        else:
            self.minStack.append(min(x, self.minStack[-1]))
        self.stack.append(x)

    def pop(self) -> None:
        if len(self.stack) != 0:
            self.stack.pop()
            self.minStack.pop()

    def top(self) -> int:
        if len(self.stack) == 0:
            return 0
        return self.stack[-1]

    def min(self) -> int:
        if len(self.minStack) == 0:
            return 0
        return self.minStack[-1]

File: Normal/2230/main.ts
class MinStack {
    stack: number[] = []
    minStack: number[] = []
    constructor() { }

    push(x: number): void {
        if (this.stack.length == 0) {
            this.minStack.push(x)
        } else {
            this.minStack.push(Math.min(x, this.minStack[this.minStack.length - 1]))
        }
        this.stack.push(x)
    }

    pop(): void {
        this.stack.pop()
        this.minStack.pop()
    }

    top(): number {
        if (this.stack.length == 0) {
            return 0
        }
        return this.stack[this.minStack.length - 1]
    }

    min(): number {
        if (this.minStack.length == 0) {
            return 0
        }
        return this.minStack[this.minStack.length - 1]
    }
}